🐛 `NameError`: name 'TYPE_CHECKING' is not defined

Ran into another test failure when trying with the latest `abipy` `develop` branch, see:

https://github.com/aiidateam/aiida-common-workflows/actions/runs/5925589563/job/16065306425?pr=314

The error is the following:

```
/root/.aiida_venvs/w2k/lib/python3.8/site-packages/abipy/tools/typing.py:12: in <module>
    if TYPE_CHECKING:  # needed to avoid circular imports
E   NameError: name 'TYPE_CHECKING' is not defined
```

The relevant lines are:

https://github.com/abinit/abipy/blob/d4dd68e000935a27fd9b4108f8ccacc7cc9cf96c/abipy/tools/typing.py#L9-L19

I'm assuming a direct import or [`TYPE_CHECKING` from the standard library module](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) should do the trick, will give it a try and open a PR.

--- a/abipy/tools/typing.py
+++ b/abipy/tools/typing.py
@@ -6,7 +6,7 @@ from __future__ import annotations
 
 import numpy as np
 
-from pymatgen.util.typing import *
+from typing import TYPE_CHECKING, Any, Union, Sequence
 
 
 if TYPE_CHECKING:  # needed to avoid circular imports

